pages/docs/core-concepts/services.mdx

```diff
@@ -0,0 +1,165 @@
+import { Callout } from "nextra-theme-docs"
+
+# Services
+
+Until now, we've only worked with Effects that have no context. In `Effect<R, E, A>`, R was always the `never` type.
+
+In this section, we'll learn how to work with Effects that have a context. We'll see how to create Effects that require a context, and how to provide that context to the Effect.
+
+## Creating a simple service interface
+
+Let's create a service that implements a random number generator.
+
+```twoslash include service-interface
+import * as Effect from "@effect/io/Effect"
+import * as Context from "@effect/data/Context";
+import { pipe } from "@effect/data/Function";
+
+
+export interface CustomRandom {
+  readonly next: () => number;
```

Comment:
this should be:

```ts
interface CustomRandom {
  readonly next: Effect.Effect<never, never, number>
}
```

as random number generation is an effect, in a pure world `() => number` can only return a constant

Reply:
Sure, may opt for a different example entirely in the future but we'll go with this for now.